source/mesh_deformation/fastscape.cc

```diff
@@ -0,0 +1,1237 @@
+                          else if (vy[index_bot+nx-1] <= 0 && vy[index_top-nx-1] < 0)
+                            {
+                              side = index_bot;
+                              op_side = index_top;
+                              jj = nx;
+                            }
+                          else
+                            continue;
+
+                          // Set top ghost node
+                          h[index_top] = h[side+jj];
+                          vx[index_top] = vx[side+jj];
+                          vy[index_top] = vy[side+jj];
+                          vz[index_top] = vz[side+jj];
+
+                          // Set bottom ghost node
+                          h[index_bot] = h[side+jj];
+                          vx[index_bot] = vx[side+jj];
+                          vy[index_bot] = vy[side+jj];
+                          vz[index_bot] = vz[side+jj];
+
+                          // Set opposing ASPECT boundary so it's periodic.
+                          h[op_side-jj] = h[side+jj];
+                          vx[op_side-jj] = vx[side+jj];
+                          vz[op_side-jj] = vz[side+jj];
+                          vy[op_side-jj] = vy[side+jj];
+                        }
+                    }    
+    }
+
+    // TODO: Give better explanations of variables and cite the fastscape documentation.
+    template <int dim>
+    void FastScape<dim>::declare_parameters(ParameterHandler &prm)
+    {
+      prm.enter_subsection ("Mesh deformation");
+      {
+        prm.enter_subsection ("Fastscape");
+        {
+          prm.declare_entry("Number of steps", "10",
+                            Patterns::Integer(),
+                            "Number of steps per ASPECT timestep");
+          prm.declare_entry("Maximum timestep", "10e3",
+                            Patterns::Double(0),
+                            "Maximum timestep for FastScape.");
+          prm.declare_entry("Vertical exaggeration", "-1",
+                            Patterns::Double(),
+                            "Vertical exaggeration for FastScape's VTK file. -1 outputs topography, basement, and sealevel.");
+          prm.declare_entry("Additional fastscape refinement", "0",
+                            Patterns::Integer(),
+                            "How many levels above ASPECT FastScape should be refined.");
+          prm.declare_entry ("Use center slice for 2d", "false",
+                             Patterns::Bool (),
+                             "If this is set to true, then a 2D model will only consider the "
+                             "center slice FastScape gives. If set to false, then aspect will"
+                             "average the inner third of what FastScape calculates.");
+          prm.declare_entry("Fastscape seed", "1000",
+                            Patterns::Integer(),
+                            "Seed used for adding an initial 0-20 m noise to FastScape topography.");
```

Comment:
I think I saw 5 m mentioned in the code?

Reply:
Yes, and this is a user set parameter now. I've updated the description here and the comment in the code.